File: reverseAnArrayUsingRecursion.java
class ReverseAnArrayUsingRecursion{
public static void main(String[] args){
    int[] n={9,8,7,6};

    reverse(n,0,n.length-1);
    for(int i=0;i<n.length;i++){
        System.out.print(n[i]+" ");
    }
}
public static void reverse(int[] a,int i,int j){
if(i>=j){
    return;
}
int temp=a[i];
a[i]=a[j];
a[j]=temp;
reverse(a,i+1,j-1);
}
}

File: reverseAnArrayUsingRecursion.js
// const reverseAnArrayUsingRecursion =(n,i,j)=>{
//     if(i>=j){
//         return;
//     }
//     [n[i],n[j]]=[n[j],n[i]];
//     reverseAnArrayUsingRecursion(n,i+1,j-1);

// };
// const n=[9,8,7,6];
// reverseAnArrayUsingRecursion(n,0,n.length-1);
// console.log(n.join(' '));

const reverseAnArrayUsingRecursion =(n,i)=>{
    if(i>=n.length/2){
        return;
    }
    [n[i],n[n.length-1-i]]=[n[n.length-1-i],n[i]];
    reverseAnArrayUsingRecursion(n,i+1);

};
const n=[9,8,7,6];
reverseAnArrayUsingRecursion(n,0);
console.log(n.join(' '));

